packages/react/src/ActionList/Item.tsx

```diff
@@ -281,20 +310,44 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
       id: itemId,
     }
 
-    const containerProps = _PrivateItemWrapper ? {role: itemRole ? 'none' : undefined} : menuItemProps
+    let containerProps
+    let wrapperProps
 
-    const wrapperProps = _PrivateItemWrapper ? menuItemProps : {}
+    if (buttonSemantics) {
+      containerProps = _PrivateItemWrapper
+        ? {role: itemRole ? 'none' : undefined, ...props}
+        : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          (listSemantics && {...menuItemProps, ...props, ref: forwardedRef}) || {}
+
+      wrapperProps = _PrivateItemWrapper
+        ? menuItemProps
+        : !listSemantics && {
+            ...menuItemProps,
+            ...props,
+            styles: merge<BetterSystemStyleObject>(styles, sxProp),
+            ref: forwardedRef,
+          }
+    } else {
+      containerProps = _PrivateItemWrapper ? {role: itemRole ? 'none' : undefined} : {...menuItemProps, ...props}
+      wrapperProps = _PrivateItemWrapper ? menuItemProps : {}
+    }
 
     return (
       <ItemContext.Provider
         value={{variant, disabled, inactive: Boolean(inactiveText), inlineDescriptionId, blockDescriptionId}}
       >
         <LiBox
-          ref={forwardedRef}
-          sx={merge<BetterSystemStyleObject>(styles, sxProp)}
+          ref={buttonSemantics || listSemantics ? forwardedRef : null}
+          sx={
+            buttonSemantics
+              ? merge<BetterSystemStyleObject>(
+                  listSemantics || _PrivateItemWrapper ? styles : listItemStyles,
+                  listSemantics || _PrivateItemWrapper ? sxProp : {},
```

Comment:
>`listSemantics || _PrivateItemWrapper ? sxProp : {},`

I am not sure if I understand this line, sorry. Does that mean we don't allow custom overrides for `listItemStyles` ? In other words, we don't want to merge `listItemStyles` with `sxProp` when `listSemantics || _PrivateItemWrapper` is false? I hope I am making sense 😅

Reply:
No worries! This is basically saying that, if the `ActionList.Item` is going to be rendered with list semantics or if it's a private item wrapper, apply the `styles` and `sxProp`, as these are the styles that you'd expect when the items are being rendered as list items only. This retains what we have right now in [`ActionList/Item.tsx`](https://github.com/primer/react/blob/main/packages/react/src/ActionList/Item.tsx#L294). 

When it's not `listSemantics` or `_PrivateItemWrapper` we need to adjust the styles slightly, [as the styles are placed on the `<button>` instead](https://github.com/primer/react/pull/4272/files#diff-21bd620cf266fd1b3eb8d73399ee032c406b9bf49eac6f598ad65c9389ddb515R278), meaning that we don't want to have duplicate styles on the `<li>` and `<button>`.

Let me know if this still makes sense, or if you have suggestions on how we can make this more readable 🤔